src/screens/Itinerary/TripItinerary.tsx

```diff
@@ -0,0 +1,91 @@
+import { RouteProp } from "@react-navigation/native";
+import { StackNavigationProp } from "@react-navigation/stack";
+import React from "react";
+import { useTranslation } from "react-i18next";
+import { ScrollView, StyleSheet, Text } from "react-native";
+import { Button, Icon } from "react-native-elements";
+import { SafeAreaView } from "react-native-safe-area-context";
+import ActivityGroup from "../../components/ActivityGroup";
+import RootStackParamList from "../../types/RootStackParamList";
+import { useGetTripQuery } from "./types/getTripQuery";
+
+type TripItineraryScreenNavigationProp = StackNavigationProp<
+    RootStackParamList,
+    "TripItinerary"
+>;
+
+type TripItineraryRouteProp = RouteProp<RootStackParamList, "TripItinerary">;
+
+type Props = {
+    navigation: TripItineraryScreenNavigationProp;
+    route: TripItineraryRouteProp;
+};
+
+export default function TripItinerary(props: Props) {
+    const { t } = useTranslation();
+
+    const { data, error, loading } = useGetTripQuery({
+        variables: { tripId: props.route.params.tripId },
+    });
+
+    if (data?.node?.__typename === "Trip") {
```

Comment:
What other possibilities are there that you need to check the type here?

Reply:
- Activity
- ActivityReaction
- Invitation
- TripMember
- TripRoutePoint
- User